source/elements/oneDPL/source/parallel_api.rst

```diff
@@ -688,5 +688,41 @@ than an element in the range being searched.
 
 The elements e of [start, end) must be partitioned with respect to the comparator used.
 
+.. code:: cpp
+
+  template <typename Policy, typename InputIt, typename OutputIt, typename UnaryOp,
+      typename UnaryPredicate>
+  OutputIt
+  transform_if(Policy&& policy, InputIt start, InputIt end, OutputIt result, UnaryOp op,
+      UnaryPredicate pred);                                                               // (1)
+
+  template <typename Policy, typename InputIt1, typename InputIt2, typename OutputIt, 
+      typename BinaryOp, typename BinaryPredicate>
+  OutputIt
+  transform_if(Policy&& policy, InputIt1 start1, InputIt1 end1, InputIt2 start2, OutputIt result,
+      BinaryOp op, BinaryPredicate pred);                                                 // (2)
+
+:code:`oneapi::dpl::transform_if` applies a given function to the elements of the input sequence(s) that
+satisfy a given predicate, and stores the result to the output. Depending on the arguments, the algorithm
+
+1.
+evaluates the unary predicate :code:`pred` for each position :code:`i` of the sequence
+:code:`[start, end)` and if :code:`pred(start[i]) == true`, it performs the unary operation
+:code:`op(start[i])` and stores the result into :code:`result[i]`. If
+:code:`pred(start[i]) == false`, the data element :code:`result[i]` is not modified from its
+initial value. The return is an iterator to the element of the output sequence
+:code:`result + std::distance(start,end)`, one past the last element which was considered
+for modification by the algorithm.
+
+2.
+evaluates the binary predicate :code:`pred` for each position :code:`i` of the sequence
+:code:`[start1, end1)` and if :code:`pred(start1[i], start2[i]) == true`, it performs the
+binary operation :code:`op(start1[i], start2[i])` and stores the result into :code:`result[i]`.
+If :code:`pred(start1[i], start2[i]) == false`, the data element :code:`result[i]` is not
+modified from its initial value. The return is an iterator to the element of the output sequence
+:code:`result + std::distance(start,end)`, one past the last element which was considered
+for modification by the algorithm.
```

Comment:
The wording for the result value is the same, so would it make sense to have it only once, outside of the numbered list? If yes, the parameter names in (1) and (2) should be aligned to `start, end`. If not, the wording here should use `start1 - end1`.

Reply:
I prefer them separated because at least for now, all the non-high level details of the algorithm are within these numbered list entries.  I think it makes more coherent sense to fully describe the algorithms in one place. 

Also, for the reason you point out that the variables are differently named, I think it is better to separate them. I would prefer to keep the variable naming as is in the prototypes, rather than to make the prototypes less readable to save some space in description.